BCStudentSoftwareDevTeam/lsf
# Turn Form Search into Supervisor Portal

Fixes issue #362 
## Changes
- **Removed** the existing **Supervisor Portal** along with **My Students** and **Department Students**.
- Changed the title of **Form Search to Supervisor Portal**.
- **Made** the **form search** bit **an accordion** that contracts when a search is processed so it only displays the search result table.
- **Removed** all instances of **index.js, index.html, and index.css**.
- Replaced formSearch.py, formSearch.js, and formSearch.html with supervisorPortal.py, etc.
- Fixed a variety of bugs in the form search mechanism.
- When the **Actions** dropdown only displays **Labor History**, the **Actions** dropdown is replaced with a button that says **Labor History** and is a direct link to the **Labor History Modal**.
- Made the student's name link to studentHistoryModal.
- Combined Form Type and Form Status into one column
- Got rid of the actions button in favor of a direct link.
- Added Department ID (ORG + ACCOUNT) to the studentHistoryModal.

## To Test
- `./reset_database from-backup`, `export FLASK_ENV=staging`, and `flask run`
- Open LSF in a window.
- Ensure that the existing Supervisor Portal tab leads you to the old Form Search page instead and that it no longer leads you to the old Supervisor Portal.
- Execute a search without selecting any options and make sure you get an error and make sure the accordion collapses.
- Execute another search with any filters selected and ensure that you can access a student's labor history modal if their form was approved, or have the ability to select an action otherwise.
- `./tests/run_tests.sh`
